workflow/controller/workflowpod.go

```diff
@@ -306,10 +306,10 @@ func (woc *wfOperationCtx) createVolumeMounts() []apiv1.VolumeMount {
 		volumeMountPodMetadata,
 	}
 	switch woc.controller.Config.ContainerRuntimeExecutor {
-	case common.ContainerRuntimeExecutorKubelet:
-		return volumeMounts
-	default:
+	case common.ContainerRuntimeExecutorDocker:
```

Comment:
Since the default should be docker the check should be the inverse:
```
case common.ContainerRuntimeExecutorKubelet , common.ContainerRuntimeExecutorK8sAPI:
    return volumeMounts
default:
    return append(volumeMounts, volumeMountDockerSock)
```

Reply:
I'll fix it now.